src/Symfony/Component/Form/Extension/Core/DataTransformer/PercentToLocalizedStringTransformer.php

```diff
@@ -129,18 +130,46 @@ public function reverseTransform($value)
             $value = str_replace(',', $decSep, $value);
         }
 
+        if (false !== strpos($value, $decSep)) {
+            $type = \NumberFormatter::TYPE_DOUBLE;
+        } else {
+            $type = PHP_INT_SIZE === 8 ? \NumberFormatter::TYPE_INT64 : \NumberFormatter::TYPE_INT32;
+        }
+
         // replace normal spaces so that the formatter can read them
-        $value = $formatter->parse(str_replace(' ', "\xc2\xa0", $value));
+        $result = $formatter->parse(str_replace(' ', "\xc2\xa0", $value), $type, $position);
 
         if (intl_is_failure($formatter->getErrorCode())) {
             throw new TransformationFailedException($formatter->getErrorMessage());
         }
 
         if (self::FRACTIONAL == $this->type) {
-            $value /= 100;
+            $result /= 100;
         }
 
-        return $value;
+        if (function_exists('mb_detect_encoding') && false !== $encoding = mb_detect_encoding($value, null, true)) {
+            $length = mb_strlen($value, $encoding);
+            $remainder = mb_substr($value, $position, $length, $encoding);
+        } else {
+            $length = strlen($value);
+            $remainder = substr($value, $position, $length);
+        }
+
+        // After parsing, position holds the index of the character where the
+        // parsing stopped
+        if ($position < $length) {
+            // Check if there are unrecognized characters at the end of the
+            // number (excluding whitespace characters)
+            $remainder = trim($remainder, " \t\n\r\0\x0b\xc2\xa0");
+
+            if ('' !== $remainder) {
+                throw new TransformationFailedException(
```

Comment:
We need tests reaching this new exception

Reply:
`PercentToLocalizedStringTransformerTest ::testDecimalSeparatorMayNotBeDotIfGroupingSeparatorIsDot()` is failing without this change on the `3.4` branch on AppVeyor: https://ci.appveyor.com/project/fabpot/symfony/build/1.0.28085#L1537

Reply:
It may be necessary to replace ` \Locale::setDefault('de_AT');` by `\Locale::setDefault('de_DE');` at  

https://github.com/symfony/symfony/blob/d3f372171597eb8a56b085e23e300ac559d5904c/src/Symfony/Component/Form/Tests/Extension/Core/DataTransformer/PercentToLocalizedStringTransformerTest.php#L147 as for https://github.com/symfony/symfony/blob/1a957100c2d4fbda247ab15bd0c85cc308d0db3c/src/Symfony/Component/Form/Tests/Extension/Core/DataTransformer/NumberToLocalizedStringTransformerTest.php#L402

See also https://github.com/symfony/symfony/commit/aa3c66e8c98ca16634126b1988b572a879f59ab1

Reply:
See also comment https://github.com/symfony/symfony/issues/19854#issuecomment-244790922

Reply:
@jakzal Why was the change in aa3c66e necessary?

Reply:
Did you see the comment https://github.com/symfony/symfony/issues/19854#issuecomment-244790922 ? You have maybe not see the previous link  

> ICU data was indeed changed for de_AT. It is taken from cldr though, which also has this bit 
> changed. So it looks like it's a bug in CLDR (unless Austria really got rid of the period as a >grouping separator).

Reply:
@aaa2000 Indeed, changing the locale to `de_DE` fixes the initial issue.